src/Symfony/Component/Routing/Tests/RouteTest.php

```diff
@@ -221,6 +221,21 @@ public function testSerializeWhenCompiled()
     }
 
     /**
+     * Tests that the compiled version does not fail when the Route refers to
+     * arbitrary classes.
```

Comment:
Does not seem to be why we are doing it. We only want to ensure that the compiled route can be different from CompiledRoute, not that you can use any object on the route definition, which is not supported.